core/shared/src/main/scala-2.x/zio/VersionSpecific.scala

```diff
@@ -1,12 +1,36 @@
 package zio
 
-import izreflect.fundamentals.reflection.Tags.Tag
+import izreflect.fundamentals.reflection.Tags._
 import izreflect.fundamentals.reflection.macrortti.{ LightTypeTag, LightTypeTagRef }
 
 private[zio] trait VersionSpecific {
 
   type TaggedType[A] = Tag[A]
-  type TagType       = LightTypeTag
+  type TaggedTypeF[F[_]] = TagK[F]
+  type TaggedTypeF2[F[_, _]] = TagKK[F]
+  type TaggedTypeF3[F[_, _, _]] = TagK3[F]
+  type TaggedTypeF4[F[_, _, _, _]] = HKTag[ { type Arg[A0, A1, A2, A3] = F[A0, A1, A2, A3] } ]
+  type TaggedTypeF5[F[_, _, _, _, _]] = HKTag[ { type Arg[A0, A1, A2, A3, A4] = F[A0, A1, A2, A3, A4] } ]
+  type TaggedTypeF6[F[_, _, _, _, _, _]] = HKTag[ { type Arg[A0, A1, A2, A3, A4, A5] = F[A0, A1, A2, A3, A4, A5] } ]
```

Comment:
I think it's not necessary to avoid syntax sugar here, the internals of HKTag aren't interesting for someone reading zio code, so you could write them in a shorter way:
```scala
type TaggedTypeF5[F[_, _, _, _, _]] = Tag.auto.T[F]
type TaggedTypeF6[F[_, _, _, _, _, _]] = Tag.auto.T[F]
```

Reply:
Thanks! Will do.